tests/Unit/ParallelAlgorithms/Events/Test_ObserveAtExtremum.cpp

```diff
@@ -0,0 +1,288 @@
+  const auto& results = MockContributeReductionData::results;
+  CHECK(results.observation_id.value() == observation_time);
+  CHECK(results.observation_id.observation_key() ==
+        expected_observation_key_for_reg);
+  CHECK(results.subfile_name == expected_subfile_name);
+  CHECK(results.time == observation_time);
+  CHECK(results.legend[0] == "Time");
+  if (extremum_type == "Max") {
+    CHECK(results.legend[1] == "Max of Var0");
+    CHECK(results.legend[2] == "Value of Var0 at Max of Var0");
+    CHECK(results.legend[3] == "Value of Var1_x at Max of Var0");
+    CHECK(results.legend[4] == "Value of Var1_y at Max of Var0");
+    CHECK(results.legend[5] == "Value of Var1_z at Max of Var0");
+    CHECK(results.reduced_vector ==
+          std::vector<double>{27.0, 27.0, 11.0, 38.0, 65.0});
+  } else {
+    CHECK(results.legend[1] == "Min of Var0");
+    CHECK(results.legend[2] == "Value of Var0 at Min of Var0");
+    CHECK(results.legend[3] == "Value of Var1_x at Min of Var0");
+    CHECK(results.legend[4] == "Value of Var1_y at Min of Var0");
+    CHECK(results.legend[5] == "Value of Var1_z at Min of Var0");
+    CHECK(results.reduced_vector ==
+          std::vector<double>{1.0, 1.0, 28.0, 12.0, 39.0});
+  }
+}
+}  // namespace
+
+SPECTRE_TEST_CASE("Unit.Evolution.ObserveAtExtremum", "[Unit][Evolution]") {
+  test<TestSectionIdTag>(
+      std::make_unique<ObserveAtExtremumEvent<TestSectionIdTag>>(
+          ObserveAtExtremumEvent<TestSectionIdTag>{
+              "reduction0", {"Var0", "Max", {"Var0", "Var1"}}}),
+      "Max", Spectral::Basis::Legendre, Spectral::Quadrature::GaussLobatto,
+      "Section0");
+  test<TestSectionIdTag>(
+      std::make_unique<ObserveAtExtremumEvent<TestSectionIdTag>>(
+          ObserveAtExtremumEvent<TestSectionIdTag>{
+              "reduction0", {"Var0", "Min", {"Var0", "Var1"}}}),
+      "Min", Spectral::Basis::Legendre, Spectral::Quadrature::GaussLobatto,
+      "Section0");
+
+  INFO("create/serialize");
+  Parallel::register_factory_classes_with_charm<Metavariables<3, void>>();
+  const auto factory_event = TestHelpers::test_creation<std::unique_ptr<Event>,
+                                                        Metavariables<3, void>>(
+      // [input_file_examples]
+      R"(
+  ObserveAtExtremum:
+    SubfileName: reduction0
+    TensorsToObserve:
+      Name: Var0
+      ExtremumType: Max
+      AdditionalData:
+      - Var0
+      - Var1
+        )");
+  // [input_file_examples]
+  auto serialized_event = serialize_and_deserialize(factory_event);
+  test<void>(std::move(serialized_event), "Max", Spectral::Basis::Legendre,
+             Spectral::Quadrature::GaussLobatto, std::nullopt);
+
+  test<void>(std::make_unique<ObserveAtExtremumEvent<void>>(
+                 ObserveAtExtremumEvent<void>{
+                     "reduction0", {"Var0", "Max", {"Var0", "Var1"}}}),
+             "Max", Spectral::Basis::FiniteDifference,
+             Spectral::Quadrature::CellCentered, std::nullopt);
+}
```

Comment:
[picking random line] Can you add a test that creates two `ReductionData` like the ones used in the event and then do `reduction_data1.combine(std::move(reduction_data2));` and check that the resulting values are what you expect? Basically this is to make sure the part that we can't easily test with the mock runtime system (reductions) is at least somewhat tested.

Reply:
Let me know if the last commit covers what you wanted.

Reply:
Yep, that's perfect, thanks! Please go ahead and squash :)